src/parachain/loans.ts

```diff
@@ -28,105 +40,425 @@ const MOCKDATA_LOAN_ASSET_KINT = {
-        return Promise.resolve(MOCKDATA_SUPPLY_POSITIONS);
+    async getLendPositionsOfAccount(accountId: AccountId): Promise<Array<LendPosition>> {
+        const marketsEntries = await this.getLoansMarketsEntries();
+        const marketsCurrencies = marketsEntries.map(([key, value]) => [key.args[0], value.unwrap().lendTokenId]);
+
+        return Promise.all(
+            marketsCurrencies.map(async ([underlyingCurrencyId, lendTokenId]) => {
+                const underlyingCurrency = await currencyIdToMonetaryCurrency(
+                    this.assetRegistryAPI,
+                    this,
+                    underlyingCurrencyId
+                );
+                return this.getLendPosition(accountId, underlyingCurrency, underlyingCurrencyId, lendTokenId);
+            })
+        );
     }
 
     getBorrowPositionsOfAccount(accountId: AccountId): Promise<Array<BorrowPosition>> {
         return Promise.resolve(MOCKDATA_BORROW_POSITIONS);
     }
 
     getLoanAssets(): Promise<TickerToData<LoanAsset>> {
+        // TODO: handle active/inactive markets
         return Promise.resolve(MOCKDATA_LOAN_ASSETS);
     }
+
+    // Check that market for given currency is added and in active state.
+    async _checkMarketState(currency: CurrencyExt, action: string): Promise<void> {
+        const underlyingCurrencyId = newCurrencyId(this.api, currency);
+        const underlyingCurrencyMarket = await this.api.query.loans.markets(underlyingCurrencyId);
+
+        if (underlyingCurrencyMarket.isNone) {
+            throw new Error(`Cannot ${action} currency that does not have market created`);
+        }
+        if (underlyingCurrencyMarket.unwrap().state.isPending) {
+            throw new Error(`Cannot ${action} currency that does not have active market.`);
+        }
+    }
+
+    async lend(accountId: AccountId, underlyingCurrency: CurrencyExt, amount: MonetaryAmount<CurrencyExt>): Promise<void> {
+        await this._checkMarketState(underlyingCurrency, "lend");
+
+        const underlyingCurrencyId = newCurrencyId(this.api, underlyingCurrency);
+        const lendTokenId = await this.getLendTokenIdFromUnderlyingCurrency(underlyingCurrency);
+        const lendExtrinsic = this.api.tx.loans.mint(underlyingCurrencyId, amount.toString(true));
+
+        // TODO: can remove after update to latest version
+        // Workaround for case when there is existing lend position
+        // used as collateral, need to batch mint -> depositCollateral
+        let extrinsicToSend = lendExtrinsic;
+
+        const underlyingCurrencyCollateralAmount = await this.api.query.loans.accountDeposits(lendTokenId, accountId);
+        if (!underlyingCurrencyCollateralAmount.isZero()) {
+            const depositAllCollateralExtrinsic = this.api.tx.loans.depositAllCollateral(underlyingCurrencyId);
+            extrinsicToSend = this.api.tx.utility.batchAll([lendExtrinsic, depositAllCollateralExtrinsic]);
+        }
```

Comment:
nice one, but this batch is no longer required

Reply:
great, I'll remove the whole check which will simplify this method greatly